test/extensions/filters/http/credential_injector/token_provider_test.cc

```diff
@@ -0,0 +1,60 @@
+#include "source/extensions/http/injected_credentials/common/secret_reader.h"
+#include "source/extensions/http/injected_credentials/oauth2/token_provider.h"
+
+#include "test/mocks/event/mocks.h"
+#include "test/mocks/server/factory_context.h"
+#include "test/mocks/thread_local/mocks.h"
+#include "test/mocks/upstream/cluster_manager.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Http {
+namespace InjectedCredentials {
+namespace OAuth2 {
```

Comment:
I think all these oauth2 tests should be placed in the `test/extensions/http/credential_injector/oauth2`?

And seems the unit tests are pretty simple (although the integration tests are great)